lib/chef/provider/git.rb

```diff
@@ -41,6 +41,20 @@ def load_current_resource
       end
 
       def define_resource_requirements
+        unless new_resource.user.nil?
+          requirements.assert(:all_actions) do |a|
+            a.assertion {
+              begin
+                require "etc" unless defined?(Etc)
+                get_dir(new_resource.user)
+              rescue ArgumentError
+                false
+              end
+            }
+            a.whyrun("User #{new_resource.user} does not exist.")
```

Comment:
```suggestion
            a.whyrun("User #{new_resource.user} does not exist, this run will fail unless it has been previously created. Assuming it would have been created.")
```